analyses/2018_12_LABBsoft_tracking_review/2019_03_toktobaeva__literature_review_web_tracking/HTTP cookies in Overscripted.py

```diff
@@ -0,0 +1,304 @@
+HTMLCanvasElement.oncontextmenu,152
+CanvasRenderingContext2D.getLineDash,146
+HTMLCanvasElement.previousSibling,139
+HTMLCanvasElement.parentElement,136
+HTMLCanvasElement.innerText,134
+HTMLCanvasElement.onkeydown,132
+HTMLCanvasElement.onkeyup,129
+HTMLCanvasElement.onkeypress,128
+HTMLCanvasElement.onblur,128
+HTMLCanvasElement.onfocus,128
+HTMLCanvasElement.onmouseleave,127
+HTMLCanvasElement.ondblclick,126
+HTMLCanvasElement.ondragenter,125
+HTMLCanvasElement.onresize,125
+HTMLCanvasElement.onpaste,125
+HTMLCanvasElement.onchange,125
+HTMLCanvasElement.oncut,125
+HTMLCanvasElement.ondragover,125
+HTMLCanvasElement.ondragleave,125
+HTMLCanvasElement.ondrop,125
+HTMLCanvasElement.onmouseenter,125
+HTMLCanvasElement.onload,125
+HTMLCanvasElement.contains,102
+HTMLCanvasElement.querySelectorAll,98
+GainNode.disconnect,77
+AudioContext.createBufferSource,70
+HTMLCanvasElement.hasChildNodes,67
+AudioContext.createBuffer,63
+AudioContext.createPanner,60
+HTMLCanvasElement.scrollLeft,60
+HTMLCanvasElement.scrollTop,60
+CanvasRenderingContext2D.__lookupSetter__,58
+CanvasRenderingContext2D.__defineSetter__,58
+HTMLCanvasElement.ondragstart,50
+HTMLCanvasElement.getClientRects,49
+HTMLCanvasElement.title,44
+HTMLCanvasElement.tabIndex,43
+RTCPeerConnection.close,43
+RTCPeerConnection.iceConnectionState,33
+AudioContext.close,32
+HTMLCanvasElement.hasAttributes,25
+HTMLCanvasElement.previousElementSibling,23
+OscillatorNode.disconnect,22
+HTMLCanvasElement.focus,22
+RTCPeerConnection.onsignalingstatechange,16
+RTCPeerConnection.oniceconnectionstatechange,16
+HTMLCanvasElement.valueOf,16
+HTMLCanvasElement.dataset,15
+HTMLCanvasElement.requestPointerLock,15
+HTMLCanvasElement.namespaceURI,13
+HTMLCanvasElement.webkitMatchesSelector,12
+HTMLCanvasElement.childElementCount,11
+HTMLCanvasElement.removeChild,8
+HTMLCanvasElement.insertBefore,8
+GainNode.numberOfOutputs,7
+HTMLCanvasElement.matches,6
+HTMLCanvasElement.outerHTML,6
+HTMLCanvasElement.appendChild,6
+AudioContext.resume,5
+AnalyserNode.getByteFrequencyData,5
+HTMLCanvasElement.clientTop,4
+HTMLCanvasElement.clientLeft,4
+HTMLCanvasElement.onwheel,4
+HTMLCanvasElement.DOCUMENT_NODE,4
+RTCPeerConnection.onaddstream,3
+AnalyserNode.channelInterpretation,3
+AnalyserNode.numberOfInputs,3
+AnalyserNode.channelCountMode,3
+AnalyserNode.numberOfOutputs,3
+AnalyserNode.channelCount,3
+HTMLCanvasElement.scrollWidth,3
+HTMLCanvasElement.scrollHeight,3
+CanvasRenderingContext2D.__proto__,3
+HTMLCanvasElement.getElementsByClassName,3
+CanvasRenderingContext2D.__defineGetter__,3
+HTMLCanvasElement.querySelector,2
+OfflineAudioContext.decodeAudioData,2
+RTCPeerConnection.createAnswer,2
+CanvasRenderingContext2D.filter,2
+AudioContext.createConvolver,1
+HTMLCanvasElement.lastChild,1
+CanvasRenderingContext2D.toString,1
+"""
+count = 0
+for line in f.split('\n'):
+    if "cookie" in line:
+        count += int(line.split(",")[1])
+        print(line[17:])
+print("Total:", count)
+
```

Comment:
There is a DRY principle in software engineering "https://en.wikipedia.org/wiki/Don't_repeat_yourself"
Any time you see large duplications like this you should be concerned and thinking how to deduplicate.

I would suggest one notebook where you document all the different queries you're looking at by reading in the original csv would be good.